src/Pages/Volunteer/Volunteer.jsx

```diff
@@ -79,19 +86,19 @@ export const Volunteer = () => {
     return watchBirdCare === "Yes" || watchFundraising === "Yes" || watchFostering === "Yes";
   };
 
-  const onCheckboxChange = (onChange) => (e) => {
+  const onCheckboxChange = (name) => (e) => {
     const newCheckboxValue = e.target.checked ? "Yes" : "No";
-    onChange(newCheckboxValue);
+    setValue(name, newCheckboxValue, { 
+      shouldValidate: true,
+      shouldDirty: true,
+      shouldTouch: true,
+    });
   };
 
-  const currentDateAsOrdinal = DateUtilities.getCurrentDateAsOrdinal();
-  const currentMonth = DateUtilities.getCurrentMonth();
-  const currentYear = DateUtilities.getCurrentFullYear();
-
   useEffect(() => {
-    const hasFormBeenSubmitted = isSubmitted;
+    const userHasClickedAnInterestCheckbox = !!(touchedFields.interested_bird_care || touchedFields.interested_fostering || touchedFields.interested_fundraising);
 
-    if (hasFormBeenSubmitted) {
+    if (userHasClickedAnInterestCheckbox) {
```

Comment:
Nested `if` statements like this can be a little hard to read. Let's change it to this pattern instead:

```
  useEffect(() => {
    const userHasClickedAnInterestCheckbox = !!(
      touchedFields.interested_bird_care ||
      touchedFields.interested_fostering ||
      touchedFields.interested_fundraising
    );

    if (!userHasClickedAnInterestCheckbox) {
      return;
    }

    if (hasSelectedOneInterest()) {
      clearErrors("has_selected_one_interest");
    } else {
      setError("has_selected_one_interest", {
        type: "manual",
        message: noInterestsSelectedMessage,
      });
    }
  }, [watchBirdCare, watchFundraising, watchFostering, setError, clearErrors]);
  ```

Reply:
Also: I noticed that your formatting doesn't match our configured settings. Do you have format on save turned on?

Reply:
Thanks for mentioning this! I had disabled my Prettier extension in an old project and hadn't enabled it again. I've enabled it again, and my files are formatting on save now.

Reply:
And great point about the nested if statements. I've made the change you recommended.